notify/sns/sns.go

```diff
@@ -41,17 +43,38 @@ type Notifier struct {
 	retrier *notify.Retrier
 }
 
+// New returns a new SNS notification handler.
+func New(c *config.SNSConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "sns", append(httpOpts, commoncfg.WithHTTP2Disabled())...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Notifier{
+		conf:    c,
+		tmpl:    t,
+		logger:  l,
+		client:  client,
+		retrier: &notify.Retrier{},
+	}, nil
+}
+
 func (n Notifier) Notify(ctx context.Context, alert ...*types.Alert) (bool, error) {
-	credentials := credentials.NewStaticCredentials(n.conf.Sigv4.AccessKey, string(n.conf.Sigv4.SecretKey), "")
-	if n.conf.Sigv4.AccessKey == "" {
-		credentials = nil
+	var(
+		err error
+		data = notify.GetTemplateData(ctx, n.tmpl, alert, n.logger)
+		tmpl = notify.TmplText(n.tmpl, data, &err)
+		creds *credentials.Credentials = nil
+	)
+	if n.conf.Sigv4.AccessKey != "" && n.conf.Sigv4.SecretKey != "" {
```

Comment:
I would suggest to have the config unmarshal failing if only one of the two is set.

Reply:
Not sure if we can do that, since it is also valid to pull the credentials from environment variables and they would be empty when we unmarshal the config yml.